packages/serverpod/lib/src/database/database_query.dart

```diff
@@ -970,12 +971,12 @@ LinkedHashMap<String, String> _gatherOrderByJoins(
 
         var queryAlias = orderByQueries[orderIndex]?.alias;
 
-        if (queryAlias == null) {
+        if (queryAlias.isNull) {
           throw _createStateErrorWithMessage(
               'Missing query alias for order by sub query with index $index.');
         }
 
-        joins[queryAlias] = _buildSubQueryJoinStatement(
+        joins[queryAlias!] = _buildSubQueryJoinStatement(
```

Comment:
Chore: Revert changes in this file.

Here is another example of when `!` is now required because the analyser cannot deduce that `queryAlias` can't be null at this point because we don't utilize `queryAlias == null`.